### PR TITLE
Implement A/B testing on landing page

### PR DESCRIPTION
**Pull request by srt32**
* Configures https://github.com/splitrb/split and adds a dashboard at `/split`.
  The dashboard is behind basic auth.
* Runs an A/B test on the auth button text on the landing page. The tests are
  defined in `config/experiments.yml`.
* Upate auth button analytics tracking to pass through the `variation`.

**Deployment**
Setting the `SPLIT_ADMIN_PASSWORD` environment variable is required.

https://trello.com/c/qmurdq4Z

### CODE REVIEW

File: config/experiments.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+auth_button:
+  alternatives:
+    - "Sign In with GitHub"
+    - "Get Started"
```
Comment (srt32) [inline config/experiments.yml]: What text would we like to test?

File: spec/requests/split_route_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  context "with auth" do
+    it "allows access" do
+      credentials = ActionController::HttpAuthentication::Basic.
+        encode_credentials(
+          "admin",
+          ENV["SPLIT_ADMIN_PASSWORD"]
+        )
+      get "/split", nil, { "HTTP_AUTHORIZATION" =>  credentials }
```
Comment (srt32) [inline spec/requests/split_route_spec.rb]: I enjoy seeing the curly braces for readability.

File: .sample.env [hunk 3]
```diff
@@ -7,6 +7,8 @@ HOUND_GITHUB_TOKEN=your_github_token
 HOUND_GITHUB_USERNAME=your_github_username
 MAX_COMMENTS=10
 QUEUE=high,medium,low
+REDIS_URL=redis://localhost:6379
```
Comment (gylaz) [inline .sample.env]: Is this required? Redis gem should have sensible defaults.

File: app/views/application/_auth_button.haml [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
+  %span= ab_test("auth_button")
 
 - content_for :analytics do
   :javascript
     var link = document.getElementsByClassName("auth")[0];
-    window.analytics.trackLink(link, "Clicked Sign In");
+    window.analytics.trackLink(
+      link,
+      "Clicked Sign In",
+      { variation: #{ ab_test("auth_button") } }
```
Comment (gylaz) [inline app/views/application/_auth_button.haml]: No spaces in `{}` when interpolating?

File: config/initializers/split.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Split.configure do |config|
+  config.experiments = YAML.load_file "config/experiments.yml"
+end
+
+Split.redis = ENV.fetch("REDIS_URL")
```
Comment (gylaz) [inline config/initializers/split.rb]: I don't think we need this. It should just pick up defaults.

File: .sample.env [hunk 6]
```diff
@@ -7,6 +7,8 @@ HOUND_GITHUB_TOKEN=your_github_token
...
 RESQUE_ADMIN_PASSWORD=password
 RESQUE_JOB_TIMEOUT=120
 SECRET_KEY_BASE=secret_key_base
+SPLIT_ADMIN_PASSWORD="split_password"
```
Comment (gylaz) [inline .sample.env]: How about dropping the quotes here for consistency?

File: .sample.env (hunk 3 above)
Comment (srt32) [inline .sample.env]: We have to set this variable on staging / production so it seems reasonable to require it here.

File: config/initializers/split.rb (hunk 5 above)
Comment (srt32) [inline config/initializers/split.rb]: You are correct but the implementation (https://github.com/splitrb/split/blob/ee1f6208fcf4506f02207d829eb2911b6898d3e5/lib/split.rb#L56) defaults to localhost so we could deploy the app without the variable and not know until something breaks later. With the setting here, the app breaks on boot if the variable is missing.

File: app/views/application/_auth_button.haml (hunk 4 above)
Comment (srt32) [inline app/views/application/_auth_button.haml]: Nice! Updated in 4a63e4f

File: .sample.env (hunk 6 above)
Comment (srt32) [inline .sample.env]: Good call! Done in 43c7939

**Comment by srt32** [comment]
@gylaz left a couple responses. Curious what you think.

**Comment by gylaz** [comment]
Looks good to me.

File: spec/requests/split_route_spec.rb (hunk 2 above)
Comment (thorncp) [inline spec/requests/split_route_spec.rb]: What about the unnecessary spacing? Two spaces after the hash rocket.

Comment (srt32) [inline spec/requests/split_route_spec.rb]: Wah! Missed that. Updated in 3f0a4e4

File: config/experiments.yml (hunk 1 above)
Comment (srt32) [inline config/experiments.yml]: Ping @frechg, did you have any ideas?

**Comment by srt32** [comment]
The build is failing but will be fixed when https://github.com/thoughtbot/hound/commit/081fcb642ccd291e58027e5d4b6d19d8522ebf2e lands.

**Pull request merged**